### PR TITLE
Remove dead goto-def/peek/ctrl-click EA impls for TS.  TS is now entirely on LSP for these experiences.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
If https://devdiv.visualstudio.com/DevDiv/_git/TypeScript-VS/pullrequest/514833 goes in first, we can remove more here.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@MariaSolOs FYI.

**Comment by CyrusNajmabadi** [comment]
@dibarbet

File: src/Tools/ExternalAccess/OmniSharp/GoToDefinition/OmniSharpFindDefinitionService.cs [hunk 1]
```diff
@@ -15,7 +15,10 @@ internal static class OmniSharpFindDefinitionService
 {
     internal static async Task<ImmutableArray<OmniSharpNavigableItem>> FindDefinitionsAsync(Document document, int position, CancellationToken cancellationToken)
     {
-        var service = document.GetRequiredLanguageService<INavigableItemsService>();
+        var service = document.GetLanguageService<INavigableItemsService>();
+        if (service is null)
+            return ImmutableArray<OmniSharpNavigableItem>.Empty;
```
Comment (CyrusNajmabadi) [inline src/Tools/ExternalAccess/OmniSharp/GoToDefinition/OmniSharpFindDefinitionService.cs:20]: just being paranoid.  this is omnisharp.  so presumably only for C#/VB.  But i'm not taking chances.

File: src/Features/LanguageServer/Protocol/Handler/Definitions/AbstractGoToDefinitionHandler.cs [hunk 2]
```diff
@@ -46,7 +46,9 @@ public AbstractGoToDefinitionHandler(IMetadataAsSourceFileService metadataAsSour
             var locations = ArrayBuilder<LSP.Location>.GetInstance();
             var position = await document.GetPositionFromLinePositionAsync(ProtocolConversions.PositionToLinePosition(request.Position), cancellationToken).ConfigureAwait(false);
 
-            var service = document.GetRequiredLanguageService<INavigableItemsService>();
+            var service = document.GetLanguageService<INavigableItemsService>();
+            if (service is null)
+                return null;
```
[the remark this answers is not in the document]
Comment (dibarbet) [reply src/Features/LanguageServer/Protocol/Handler/Definitions/AbstractGoToDefinitionHandler.cs:51]: I kind of prefer throwing actually.  It isn't expected that we would generally hit this, and if we do we'll get errors that tell us what is going wrong.  It won't take down the LSP server or anything.

**Review by dibarbet — APPROVED** [review]

File: src/Tools/ExternalAccess/OmniSharp/GoToDefinition/OmniSharpFindDefinitionService.cs (hunk 1 above)
Comment (333fred) [reply src/Tools/ExternalAccess/OmniSharp/GoToDefinition/OmniSharpFindDefinitionService.cs:20]: Note even VB, just C#. But I'm ok with paranoia.

**Pull request merged**